common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -369,32 +368,33 @@ public Object run() {
     }
 
     static boolean isExplicitNoUnsafe() {
-        return IS_EXPLICIT_NO_UNSAFE;
+        return EXPLICIT_NO_UNSAFE_CAUSE == null;
     }
 
-    private static boolean explicitNoUnsafe0() {
+    private static Throwable explicitNoUnsafeCause0() {
         final boolean noUnsafe = SystemPropertyUtil.getBoolean("io.netty.noUnsafe", false);
         logger.debug("-Dio.netty.noUnsafe: {}", noUnsafe);
 
         if (noUnsafe) {
             logger.debug("sun.misc.Unsafe: unavailable (io.netty.noUnsafe)");
-            return true;
+            return new UnsupportedOperationException("sun.misc.Unsafe: unavailable (io.netty.noUnsafe)");
         }
 
         // Legacy properties
-        boolean tryUnsafe;
+        String unsafePropName;
         if (SystemPropertyUtil.contains("io.netty.tryUnsafe")) {
-            tryUnsafe = SystemPropertyUtil.getBoolean("io.netty.tryUnsafe", true);
+            unsafePropName = "io.netty.tryUnsafe";
         } else {
-            tryUnsafe = SystemPropertyUtil.getBoolean("org.jboss.netty.tryUnsafe", true);
+            unsafePropName = "org.jboss.netty.tryUnsafe";
         }
 
-        if (!tryUnsafe) {
-            logger.debug("sun.misc.Unsafe: unavailable (io.netty.tryUnsafe/org.jboss.netty.tryUnsafe)");
-            return true;
+        if (!SystemPropertyUtil.getBoolean(unsafePropName, true)) {
+            String msg = String.format("sun.misc.Unsafe: unavailable (%s)", unsafePropName);
```

Comment:
Just use string concatation ? I think there is no much win in use String.format at all.

Reply:
Done.